airflow/docker_image.go

```diff
@@ -276,6 +275,92 @@ func (d *DockerImage) TagLocalImage(localImage string) error {
 	return nil
 }
 
+func (d *DockerImage) RunTest(dagID, envFile, settingsFile, startDate, containerName string, taskLogs bool) error {
+	// delete container
+	stdout := os.Stdout
+	stderr := os.Stderr
+	err := cmdExec(DockerCmd, nil, nil, "rm", "astro-run")
+	if err != nil {
+		log.Debug(err)
+	}
+	var args []string
+	if containerName != "" {
+		args = []string{
+			"exec",
+			"-t",
+			containerName,
+		}
+	}
+	// docker exec
+	if containerName == "" {
+		// convert Settings file to variables and connection yaml
+		err = settings.FileToConnectionYAML(settingsFile)
+		if err != nil {
+			log.Debug(err)
+		}
+		args = []string{
+			"run",
+			"-t",
+			"--name",
+			"astro-run",
+			"-v",
+			config.WorkingPath + "/dags:/usr/local/airflow/dags:ro",
+			"-v",
+			config.WorkingPath + "/" + settingsFile + ":/usr/local/" + settingsFile,
+			"-v",
+			config.WorkingPath + "/plugins:/usr/local/airflow/plugins:z",
+			"-v",
+			config.WorkingPath + "/include:/usr/local/airflow/include:z",
+			"-v",
+			config.WorkingPath + "/variables.yaml:/usr/local/airflow/variables.yaml:z",
+			"-v",
+			config.WorkingPath + "/connections.yaml:/usr/local/airflow/connections.yaml:z",
+		}
+		// if env file exists append it to args
+		fileExist, err := util.Exists(config.WorkingPath + "/" + envFile)
+		if err != nil {
+			log.Debug(err)
+		}
+		if fileExist {
+			args = append(args, []string{"--env-file", envFile}...)
+		}
+		args = append(args, []string{d.imageName}...)
+	}
+	cmdArgs := []string{
+		"flow",
+		"test-dag",
+		"./dags/",
+		dagID,
+		"/usr/local/airflow/connections.yaml",
+		"/usr/local/airflow/variables.yaml",
+	}
+	args = append(args, cmdArgs...)
+
+	fmt.Println("\nStarting a DAG run for " + dagID + "...")
+	fmt.Println("\nLoading DAGS...")
+
+	cmdErr := cmdExec(DockerCmd, stdout, stderr, args...)
+	if cmdErr != nil {
+		log.Debug(cmdErr)
```

Comment:
shouldn't we return the error in case we are not able to run the command? or do we want to hide any/all docker-related errors in this command from the users?

Reply:
+1

Reply:
This error "cmdErr" is returned at the end of the command I just want to make sure that the container is deleted if the command fails